src/tests/test_tls_stream_integration.cpp

```diff
@@ -371,7 +454,17 @@ class Synchronous_Test : public TestBase
 
       void run(const error_code&)
          {
-         m_client_thread = std::thread(std::bind(&Synchronous_Test::run_synchronous_client, this));
+         m_client_thread = std::thread([this]
+            {
+            try
+               { this->run_synchronous_client(); }
+            catch(const std::exception& ex)
+               { m_result.test_failure(std::string("sync client failed with: ") + ex.what()); }
+            catch(const boost::exception&)
+               { m_result.test_failure("sync client failed with boost exception"); }
```

Comment:
Why not record information about the boost exception like it is done for the std one?

Reply:
Frankly: Because I couldn't figure out how to get the error message from boost extensions. O.o

Reply:
There seems to exist a [`diagnostic_information`](https://www.boost.org/doc/libs/1_81_0/libs/exception/doc/diagnostic_information.html) function. But would be ok for me to also leave as is